tests/unittest.py

```diff
@@ -334,6 +335,12 @@ def setUp(self) -> None:
         calling the prepare function.
         """
         self.reactor, self.clock = get_clock()
+
+        # The home server will start connecting places as soon as it starts,
+        # so we must update some fake DNS entries early.
+        if HomeserverTestCase.lookups:
+            self.reactor.lookups.update(HomeserverTestCase.lookups)
```

Comment:
I suspect these should be `self.lookups`?

```suggestion
        if self.lookups:
            self.reactor.lookups.update(self.lookups)
```

Reply:
No, this is fine like it is, it is propagating these lookups to the reactor, which uses that dictionary in the `FakeResolver`, see: https://github.com/matrix-org/synapse/blob/84f441f88f51d3f94e1616e1e5507df0dadb6de8/tests/server.py#L468